File: Behavioral/Lab15/Typescript/src/Commands/abstract_command.ts
import { ICommand } from "../Interfaces/command";

export abstract class AbstractCommand implements ICommand {
    protected _params: string[];
    constructor(...args: string[]) {
        this._params = args;
    }

    public get Params(): string[] {
        return this._params;
    }

    public set Params(value: string[]) {
        this._params = value;
    }

    public abstract execute(): string;
}

File: Behavioral/Lab15/C#/Commands/AbstractCommand.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace C_.TemplateMethods
{
    public abstract class AbstractCommand : Interfaces.ICommand
    {
        protected string[] _params;
        public AbstractCommand(params string[] args){
            _params = args;
        }
        public string[] Params{get; set;}

        public abstract string Execute();
    }
}
